tests/conformance/state/state.go

```diff
@@ -298,6 +298,45 @@ func ConformanceTests(t *testing.T, props map[string]string, statestore state.St
 		})
 	}
 
+	if config.HasOperation("get") && config.HasOperation("set") {
+		t.Run("read, write, read output match", func(t *testing.T) {
+			for _, scenario := range scenarios {
+				if !scenario.bulkOnly && !scenario.transactionOnly {
+					t.Logf("Retrieving value for %s", scenario.key)
+					req := &state.GetRequest{
+						Key: scenario.key,
+					}
+					if len(scenario.contentType) != 0 {
+						req.Metadata = map[string]string{metadata.ContentType: scenario.contentType}
+					}
+					res, err := statestore.Get(req)
+					assert.Nil(t, err)
+					assertEquals(t, scenario.value, res)
+					t.Logf("Rewriting retrieved value for %s", scenario.key)
+					setReq := &state.SetRequest{
+						Key:   scenario.key,
+						Value: res,
+					}
+					if len(scenario.contentType) != 0 {
+						req.Metadata = map[string]string{metadata.ContentType: scenario.contentType}
+					}
+					err2 := statestore.Set(setReq)
+					assert.Nil(t, err2)
+					t.Logf("Retrieving rewritten value for key %s", scenario.key)
+					req2 := &state.GetRequest{
+						Key: scenario.key,
+					}
+					if len(scenario.contentType) != 0 {
+						req.Metadata = map[string]string{metadata.ContentType: scenario.contentType}
+					}
+					res2, err := statestore.Get(req2)
+					assert.Nil(t, err)
+					assertEquals(t, res, res2)
```

Comment:
In `assertEquals` method, the first argument after `t`, here it is `res`, is of type `interface{}` in the method definition whose type is used to properly unmarshal the field `Data` from the later argument, here `res2` which is actually the GetResponse. Again the issue here is `res2.Data` is actual data that we are interested in, and `res` is another GetResponse object and not the actual data that is compared against. 
So this should be `assertEquals(t, scenario.value, res2)`.